roipoly/roipoly.py

```diff
@@ -69,15 +74,17 @@ def __init__(self, fig=None, ax=None, color='b',
         self.__cid2 = self.fig.canvas.mpl_connect(
             'button_press_event', self.__button_press_callback)
 
+        # Determine the type of environment
+        self.interactive = sys.flags.interactive or USING_SPYDER
+
         if show_fig:
             self.show_figure()
 
-    @staticmethod
-    def show_figure():
-        if sys.flags.interactive:
+    def show_figure(self):
+        while (not self.completed) and self.interactive:
             plt.show(block=False)
-        else:
-            plt.show(block=True)
+            plt.pause(0.1)
```

Comment:
I think I found a simpler solution that worked for me (ubuntu): Just chaning the graphics backend of ipython in spyder to `automatic`  as described here https://stackoverflow.com/questions/23585126/how-do-i-get-interactive-plots-again-in-spyder-ipython-matplotlib
![image](https://user-images.githubusercontent.com/7914340/74108829-2f50ac00-4b7e-11ea-8456-e13033f76910.png)

Can you confirm that this also works on your windows setup? If yes, I'd prefer just adding this to the readme instead of having the workaround the code.

Reply:
> 
> 
> I think I found a simpler solution that worked for me (ubuntu): Just chaning the graphics backend of ipython in spyder to `automatic` as described here ..
> 
> Can you confirm that this also works on your windows setup? If yes, I'd prefer just adding this to the readme instead of having the workaround in the code.

This was the first thing I tried before investigating the problem - I tried every backend (including deactivating support for graphics). It does not work on my setup (Windows 10, spyder 4). Downgrading to spyder 3, as you're using in the screenshot, also did not fix the issue.

However, on another machine with Windows 7 and spyder 3 (same python version) it works correctly with the automatic backend. Upgrading to spyder 4 on this machine caused the problem to occur. All in all a confusing situation.

Reply:
Oh god, that sounds like a nightmare 😂 I will try to test on a Mac with different spyder versions this weekend